lib/puppet_x/certregen/crl.rb

```diff
@@ -0,0 +1,55 @@
+require 'fileutils'
+require 'openssl'
+
+module PuppetX
+  module Certregen
+    # @api private
+    # @see {Puppet::SSL::CertificateRevocationList'
+    module CRL
+      module_function
+
+      FIVE_YEARS = 5 * 365*24*60*60
+
+      def refresh(ca)
+        crl = ca.crl
+        crl_content = crl.content
+        update_to_next_crl_number(crl_content)
+        update_valid_time_range_to_start_at(crl_content, Time.now)
+        sign_with(crl_content, ca.host.key.content)
+        Puppet::SSL::CertificateRevocationList.indirection.save(crl)
+        FileUtils.cp(Puppet[:cacrl], Puppet[:hostcrl])
+      end
+
+      # @api private
+      def update_valid_time_range_to_start_at(crl_content, time)
+        # The CRL is not valid if the time of checking == the time of last_update.
+        # So to have it valid right now we need to say that it was updated one second ago.
+        crl_content.last_update = time - 1
+        crl_content.next_update = time + FIVE_YEARS
```

Comment:
Good catch. For that it's worth `ca_ttl` is a misnomer as it's the ttl for all signed certificates, so let's not overload it more.